File: solution/058LengthOfLastWord/058LengthOfLastWord.go
package main

func lengthOfLastWord(s string) int {
	n := len(s)
	j := n - 1
	for j >= 0 && s[j] == ' ' {
		j--
	}
	i := j
	for j >= 0 && s[j] != ' ' {
		j--
	}
	return i - j
}

func main() {

}

File: solution/058LengthOfLastWord/058LengthOfLastWord.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let s = s.trim_end();
        let mut chars = s.chars().rev();
        let mut length = 0;
        while let Some(ch) = chars.next() {
            if ch == ' ' {
                break;
            }    
            length += 1;
        }
        
        length
    }
}
